spec/publishing_api_response_spec.rb

```diff
@@ -372,6 +377,69 @@
     end
   end
 
+  context "homepage content item with a level_one_taxon and a child" do
```

Comment:
@1pretz1 You're only using homepage_taxon in one test, so it doesn't need to be in a `let` block. Generally variables are created in `let` and `before` blocks when they needed in multiple tests.

It will make the test easier to follow if everything from the `let` is moved into the test, and `homepage_taxon` is set to just the result of calling `from_content_id`.

e.g.

```
  context "homepage content item with a level_one_taxon and a child" do
    it "parses each level of taxons from home page" do
      child_for_level_one_taxon = {
        "content_id" => "84aadc14-9bca-40d9-abb4-4f21f9792a05",
        "base_path" => "/transport_child",
        "title" => "Transport child",
        "details" => {
          "internal_name" => "TC 1",
        },
        "links" => {}
      }

      level_one_taxon = {
        "content_id" => "a4038b29-b332-4f13-98b1-1c9709e216bc",
        "base_path" => "/transport/all",
        "title" => "Transport",
        "details" => {
          "internal_name" => "Transport",
        },
        "links" => {
          "child_taxons" => [child_for_level_one_taxon],
          "root_taxon" => [content_item]
        }
      }

      expanded_links = {
        "expanded_links" => {
          "level_one_taxons" => [level_one_taxon],
        }
      }

      expanded_links_2 = {
        "expanded_links" => {
          "child_taxons" => [child_for_level_one_taxon]
        }
      }

      content_item = {
        "content_id" => "f3bbdec2-0e62-4520-a7fd-6ffd5d36e03a",
        "base_path" => "/",
        "title" => "GOV.UK homepage",
        "details" => {}
      }

      allow(publishing_api).to receive(:get_content).with('f3bbdec2-0e62-4520-a7fd-6ffd5d36e03a').and_return(content_item)
      allow(publishing_api).to receive(:get_expanded_links).with('f3bbdec2-0e62-4520-a7fd-6ffd5d36e03a').and_return(expanded_links)
      allow(publishing_api).to receive(:get_expanded_links).with('a4038b29-b332-4f13-98b1-1c9709e216bc').and_return(expanded_links_2)

      homepage_taxon = GovukTaxonomyHelpers::LinkedContentItem.from_content_id(
        content_id: content_item['content_id'],
        publishing_api: publishing_api
      )

      expect(homepage_taxon.title).to eq("GOV.UK homepage")
      expect(homepage_taxon.parent).to eq(nil)
      expect(homepage_taxon.children.map(&:title)).to eq(["Transport"])
      expect(homepage_taxon.descendants.map(&:title)).to eq(["Transport", "Transport child"])
      expect(homepage_taxon.children.first.children.first.title).to eq("Transport child")
    end
  end
```

I'd also consider renaming `content_item` to `root_taxon` so it's a bit easier to picture what the homepage_taxon content_item looks like.

The same approach will work from the "minimal responses with missing links and details hashes"  test below as well.

If you're interested, there is a bit of discussion [here](https://stackoverflow.com/questions/5974360/rspec-what-is-the-difference-between-let-and-a-before-block) about when to use a `let` or a `before` block.